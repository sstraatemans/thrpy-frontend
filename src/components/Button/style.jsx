import styled, { keyframes } from "styled-components";

export const StyledButton = styled.a`
  position: relative;
  display: inline-block;
  padding: 0 2rem;
  overflow: hidden;
  z-index: 1;

  font-family: ${props => props.theme.fontFamily};
  text-decoration: none;
  color: #fff;
  background-color: ${props => props.theme.primaryColor};
  text-align: center;
  letter-spacing: 0.5px;
  transition: 0.2s ease-out;
  cursor: pointer;
  border: none;
  border-radius: 2px;
  height: 36px;
  line-height: 36px;
  text-transform: uppercase;
  user-select: none;
  box-shadow: props => props.theme.boxShadow</1>
  transition: .3s ease-out;
}


`;

export const FlatButton = StyledButton.extend`
  box-shadow: none;
  background-color: transparent;
  color: #343434;
`;
