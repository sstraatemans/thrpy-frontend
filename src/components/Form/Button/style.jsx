import styled, { keyframes } from "styled-components";

const iconRotate = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(359deg);
  }
`;

export const ButtonBar = styled.div`
  margin-top: 20px;
  text-align: right;
`;

export const StyledButtonWrapper = styled.div`
  display: inline-block;
  &:after {
    position: relative;
    display: inline-block;
    right: 33px;
    top: -1px;
    content: "";
    width: 10px;
    height: 10px;
    border: 2px solid white;
    border-left: 0;
    border-top: 0;
    transform: rotate(-45deg);
    transition: all 0.5s;
  }

  ${props =>
    props.isLoading &&
    `
    &:after {
      right: 30px;
      border-radius: 50px;
      transition: all 0.5s;
      animation: ${iconRotate} 2s infinite linear;
    }
  `} &:hover {
    &:after {
      right: 30px;
    }

    input {
      background: rgba(234, 98, 49, 1); /* Old browsers */
      background: linear-gradient(
        to right,
        rgba(237, 130, 33, 1) 72%,
        rgba(234, 98, 49, 1) 100%
      );
    }
  }
`;

export const StyledButton = styled.input`
  width: 270px;
  padding: 20px 30px;
  border: 0;
  border-radius: 30px;
  text-align: left;

  font-size: 16px;
  font-weight: 100;
  text-transform: uppercase;
  color: white;

  cursor: pointer;
  background: rgb(237, 130, 33); /* Old browsers */
  background: linear-gradient(
    to right,
    rgba(237, 130, 33, 1) 0%,
    rgba(234, 98, 49, 1) 100%
  );
`;
