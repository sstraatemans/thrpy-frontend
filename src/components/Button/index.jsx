// @flow
import React from "react";
import { StyledButton, FlatButton } from "./style";

export const ButtonTypes = {
  Normal: "Normal",
  Flat: "Flat"
};
type ButtonType = $Keys<typeof ButtonTypes>;

type Props = {
  onClick: Function,
  style?: ButtonType,
  children: any
};

const defaultProps = { style: "Normal" };

const Button = (props: Props) => {
  if (props.style === ButtonTypes.Flat) {
    return <FlatButton onClick={props.onClick}>{props.children}</FlatButton>;
  }
  return <StyledButton onClick={props.onClick}>{props.children}</StyledButton>;
};

export default Button;
