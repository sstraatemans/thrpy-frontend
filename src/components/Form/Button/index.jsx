// @flow
import React from "react";
import { StyledButton, StyledButtonWrapper } from "./style";

type Props = {
  isLoading: boolean,
  label: string
};

const Button = (props: Props) => {
  return (
    <StyledButtonWrapper isLoading={props.isLoading ? "true" : ""}>
      <StyledButton type="submit" value={props.label} />
    </StyledButtonWrapper>
  );
};

export default Button;
