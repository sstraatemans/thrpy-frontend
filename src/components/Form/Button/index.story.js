import React from "react";

import { storiesOf } from "@storybook/react";
import { specs, describe, it } from "storybook-addon-specifications";
import { mount } from "enzyme";
import expect from "chai";
import { action } from "@storybook/addon-actions";
import Button from "./";

const stories = storiesOf("Form/Button", module);

stories.add("Hello World", function() {
  const story = <Button>Hello World</Button>;

  // specs(() =>
  //   describe("Hello World", function() {
  //     it("Should have the Hello World label", function() {
  //       let output = mount(story);
  //       console.log(output);
  //       expect(output.text()).toContain("Hello World");
  //     });
  //   })
  // );

  return story;
});

stories.add("isloading", () => <Button isLoading="true">Loader</Button>);
