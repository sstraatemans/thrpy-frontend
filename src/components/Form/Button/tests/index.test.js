import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Button from "./../";

describe("<Button />", () => {
  it("render component <Button>", () => {
    const wrapper = mount(<Button />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("sets loading param", () => {
    let props = {
      isLoading: true
    };
    const renderedComponent = shallow(<Button {...props} />);
    expect(renderedComponent.prop("isLoading")).toEqual("true");
  });

  it("do not set loading param", () => {
    let props = {};
    const renderedComponent = shallow(<Button {...props} />);

    expect(renderedComponent.prop("isLoading")).toEqual("");
  });
});
