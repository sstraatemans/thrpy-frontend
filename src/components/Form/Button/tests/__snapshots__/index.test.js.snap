exports[`<Button /> render component <Button> 1`] = `
<Button>
  <styled.div
    isLoading="">
    <div
      className="sc-bwzfXH gZEvPN">
      <styled.input
        type="submit"
        value={undefined}>
        <input
          className="sc-htpNat jJfBff"
          type="submit"
          value={undefined} />
      </styled.input>
    </div>
  </styled.div>
</Button>
`;
